src/modules/cpu/kernel/color_cast.hpp

```diff
@@ -0,0 +1,1211 @@
+#include "rppdefs.h"
+#include "cpu/rpp_cpu_simd.hpp"
+#include "cpu/rpp_cpu_common.hpp"
+
+RppStatus color_cast_u8_u8_host_tensor(Rpp8u *srcPtr,
+                                       RpptDescPtr srcDescPtr,
+                                       Rpp8u *dstPtr,
+                                       RpptDescPtr dstDescPtr,
+                                       RpptRGB *rgbTensor,
+                                       Rpp32f *alphaTensor,
+                                       RpptROIPtr roiTensorPtrSrc,
+                                       RpptRoiType roiType,
+                                       RppLayoutParams layoutParams)
+{
+    RpptROI roiDefault;
+    RpptROIPtr roiPtrDefault;
+    roiPtrDefault = &roiDefault;
+    roiPtrDefault->xywhROI.xy.x = 0;
+    roiPtrDefault->xywhROI.xy.y = 0;
+    roiPtrDefault->xywhROI.roiWidth = srcDescPtr->w;
+    roiPtrDefault->xywhROI.roiHeight = srcDescPtr->h;
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(dstDescPtr->n)
+    for(int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        RpptROI roi;
+        RpptROIPtr roiPtr;
+
+        if (&roiTensorPtrSrc[batchCount] == NULL)
```

Comment:
This code block for checking roi is repeated in all the functions. We need to discuss to condense it and put it in a macro/function

Reply:
Done. Condensed that code block in all instances to a common function call.